src/Kestrel.Core/Internal/Http/Http1Connection.cs

```diff
@@ -413,67 +413,81 @@ protected override string CreateRequestId()
         protected override MessageBody CreateMessageBody()
             => Http1MessageBody.For(_httpVersion, HttpRequestHeaders, this);
 
-        protected override async Task<bool> ParseRequestAsync()
+        protected override void BeginRequestProcessing()
```

Comment:
👍  It does feel better to not rely on the state always transitioning on the first iteration of the loop.

The other idea didn't work because of [this CancelPendingRead in Http1MessageBody.StopAsync()](https://github.com/aspnet/KestrelHttpServer/blob/dev/src/Kestrel.Core/Internal/Http/Http1MessageBody.cs#L129). I forgot that this also called CancelPendingRead.